### PR TITLE
Learning/pbr 14 checkout wireframe

### PR DESCRIPTION
**Pull request by ZFenton**
<h2>Overview</h2>
Cart functionality added along with donate page (my bad, should have been separate branch). Method extracted portions of my code as to not have widgets spanning the whole editor. 

Things to note: The list still isn't scrollable at the moment, and adding more than two donations will make the list run out of pixels; displaying the zebra striped error thingy.

<h2>Process</h2>
Not sure what to put here so I basically did tutorials and codelabs on building scrolling lists, and then used bits of Taylors list code to come up with a checkout that has a "banner" so to speak at the top that lists the total and has the donate button. Clicking the donate button directs you to the donation page and adding (any amount at this point) adds a $10.00 donation item to the list.

<ul>
<li>At least 2 team members must approve all pull requests (PR)</li>
<li>Test notes:</li>
  <ul>
  <li>pretend you have no prior knowledge of the app</li>
  <li>make sure navigation works</li>
  <li>ensure clicking 'donation' directs you to the donation page</li>
  <li>add any number in the text field and click donate; make sure donation is added to list</li>
</ul>
</ul>

### CODE REVIEW

File: pbr/pubspec.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^0.1.2
+  english_words: ^3.1.0
```
Comment (ZFenton) [inline pbr/pubspec.yaml]: ignore  this. left over from a codelab and forgot to remove.

File: pbr/lib/uiComponents.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import 'package:flutter/material.dart';
+
```
Comment (ZFenton) [inline pbr/lib/uiComponents.dart]: If no one is privy to the idea of having a dart file for universal components, then I can easily move this back into donation file

**Comment by tadowdy** [comment]
I'm getting a new error message when i run it when the iPhone simulator. 
`flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following message was thrown during layout:
flutter: A RenderFlex overflowed by 8.0 pixels on the right.
flutter:
flutter: The overflowing RenderFlex has an orientation of Axis.horizontal.
flutter: The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
flutter: black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
flutter: Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
flutter: RenderFlex to fit within the available space instead of being sized to their natural size.
flutter: This is considered an error condition because it indicates that there is content that cannot be
flutter: seen. If the content is legitimately bigger than the available space, consider clipping it with a
flutter: ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
flutter: like a ListView.
flutter: The specific RenderFlex in question is:
flutter:   RenderFlex#2ecb7 relayoutBoundary=up2 OVERFLOWING
flutter:   creator: Row ← Column ← Container ← CartPageBody ← MediaQuery ← LayoutId-[<_ScaffoldSlot.body>] ←
flutter:   CustomMultiChildLayout ← AnimatedBuilder ← DefaultTextStyle ← AnimatedDefaultTextStyle ←
flutter:   _InkFeatures-[GlobalKey#41ebd ink renderer] ← NotificationListener<LayoutChangedNotification> ← ⋯
flutter:   parentData: offset=Offset(0.0, 0.0); flex=null; fit=null (can use size)
flutter:   constraints: BoxConstraints(0.0<=w<=414.0, 0.0<=h<=Infinity)
flutter:   size: Size(414.0, 48.0)
flutter:   direction: horizontal
flutter:   mainAxisAlignment: start
flutter:   mainAxisSize: max
flutter:   crossAxisAlignment: center
flutter:   textDirection: ltr
flutter:   verticalDirection: down
flutter: ◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
`

**Comment by tadowdy** [comment]
Changes in PR #15. Closing this PR.

**Pull request closed without merging**